### PR TITLE
Optimizes atmos init by 20%

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Each time we intialized a turf's atmos, we checked all the turfs around it to see if they were different. This meant each pair of turfs talked to each other twice.

If we instead do the comparing in a second loop, we can use current_cycle to ensure we only compare a pair once. This saves 0.5 seconds of atmos init.

## Why It's Good For The Game

Speed

### CODE REVIEW

File: code/controllers/subsystem/air.dm [hunk 1]
```diff
@@ -538,15 +538,35 @@ SUBSYSTEM_DEF(air)
...
 		// This way we can make setting up adjacent turfs O(n) rather then O(n^2)
 		T.Initalize_Atmos(time)
+		// We assert that we'll only get open turfs here
+		difference_check += T
 		if(CHECK_TICK)
 			time--
 
+	// Now we're gonna compare for differences
+	// Taking advantage of current cycle being set to negative before this run to do A->B B->A prevention
+	for(var/turf/open/T as anything in difference_check)
```
Comment (Mothblocks) [inline code/controllers/subsystem/air.dm:555]: No `T`

File: code/controllers/subsystem/air.dm [hunk 2]
```diff
@@ -538,15 +538,35 @@ SUBSYSTEM_DEF(air)
...
+		T.current_cycle = 0
+		for(var/turf/open/enemy_tile as anything in T.atmos_adjacent_turfs)
+			// If it's already been processed, then it's already talked to us
+			if(enemy_tile.current_cycle == 0)
+				continue
+			// .air instead of .return_air() because we can guarentee that the proc won't do anything
+			if(T.air.compare(enemy_tile.air))
+				//testing("Active turf found. Return value of compare(): [T.air.compare(enemy_tile.air)]")
```
Comment (Mothblocks) [inline code/controllers/subsystem/air.dm:563]: I don't think we should have this commented code

**Review by Mothblocks — APPROVED** [review]

File: code/controllers/subsystem/air.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/controllers/subsystem/air.dm:555]: sure

File: code/controllers/subsystem/air.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/controllers/subsystem/air.dm:563]: it's soul you can't take him from me https://github.com/tgstation/tgstation/pull/16026

**Pull request merged**